client/src/Components/ProductPage/ProductPage.js

```diff
@@ -0,0 +1,73 @@
+import { useNavigate, useParams } from "react-router";
+import { useCart } from "../../cart-context/cart-context";
+import { AddtoCart } from "../ProductCard/AddtoCart";
+import { AddMore } from "../ProductCard/AddMore";
+import { useAuth } from "../../Auth/auth-context";
+import callServer from "../../api_calls/axios";
```

Comment:
i liked that you have made a separate function to call your apis....i might do that too for my project!!